src/util/providers/AuthProvider.tsx

```diff
@@ -68,17 +71,32 @@ export const useAuthContext = (): AuthContext => {
     authenticate,
     signout,
     signup,
+    error,
   };
 };
 
 export const timeout = (ms: number) =>
   new Promise((res) => setTimeout(res, ms));
 
+const usePrevious = (value: string) => {
+  const ref = useRef<string>();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+  return ref.current;
+}
+
 const AuthProvider = ({ children }: AuthProviderProps) => {
   const history = useHistory();
   const [currentUser, setCurrentUser] = useState<User>(defaultCurrentUser);
+  const [error, setError] = useState('');
 
+  const prevError = usePrevious(error);
   useEffect(() => {
+    if (prevError) {
```

Comment:
Let's set up a separate `useEffect` hook for the `setError('')` to separate the error clearing from the session restoration.

https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns